src/test/java/io/vertx/cassandra/SharedTest.java

```diff
@@ -0,0 +1,45 @@
+package io.vertx.cassandra;
+
+import com.datastax.driver.core.Row;
+import io.vertx.core.Future;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+@RunWith(VertxUnitRunner.class)
+public class SharedTest extends CassandraServiceBase {
+
+  @Test
+  public void executeQueryFormLookUp(TestContext context) {
```

Comment:
What does this test does exactly? It doesn't seem to verify the same session is used.

Reply:
Initially, we are creating a shared client, then we, using the client when connecting to the test Cassandra. After that, we are doing the lookup (not creating the client, but getting already created and connected client) and executing a query with it and disconnecting(after another lookup).

If the test passed successfully it would mean, that for these 3 operations(connect, query execution, disconnect) only a single client had been used.

Reply:
Ha, you mean if it was not the same client, the second phase would fail because the client is not connected yet?

Reply:
Sure

Reply:
👍